cypress/e2e/opening-hours-editor.cy.ts

```diff
@@ -0,0 +1,298 @@
+const branchTitle = "Test branch";
+const formattedSearchString = branchTitle.toLowerCase().replace(/ /g, "+");
+
+enum OpeningHourCategories {
+  Opening = "Åbent",
+  CitizenService = "Borgerservice",
+  WithService = "Med betjening",
+  SelfService = "Selvbetjening",
+  PhoneTime = "Telefontid",
+}
+
+type OpeningHourFormType = {
+  openingHourCategory: OpeningHourCategories;
+  start: `${number}:${number}`;
+  end: `${number}:${number}`;
+};
+
+const createTestBranchAndVisitOpeningHoursAdmin = () => {
+  cy.drupalLoginAndVisit("/node/add/branch");
+  cy.get("#edit-title-0-value").type(branchTitle);
+  cy.get('button[title="Show all Paragraphs"]').click();
+  cy.get('button[value="Opening Hours"]').click({
+    multiple: true,
+    force: true,
+  });
+  cy.get("#edit-field-address-0-address-address-line1")
+    .type("Example Street", { force: true })
+    .should("have.value", "Example Street");
+  cy.get("#edit-field-address-0-address-postal-code")
+    .type("1234", { force: true })
+    .should("have.value", "1234");
+  cy.get("#edit-field-address-0-address-locality")
+    .type("Example City", { force: true })
+    .should("have.value", "Example City");
+  cy.get('input[value="Save"]').click();
+  cy.get('a[href^="/node/"][href$="/edit"]').click({ force: true });
+  cy.get('a[href*="/edit/opening-hours"]').click();
+  // Save the URL for the admin page and the page itself for later use
+  cy.url().then((url) => {
+    Cypress.env("adminUrl", url);
+    const pageUrl = url.replace("/edit/opening-hours", "");
+    Cypress.env("pageUrl", pageUrl);
+  });
+};
+const deleteAllTestBranchesIfExists = () => {
+  cy.drupalLogin();
+  cy.visit(
+    `/admin/content?title=${formattedSearchString}&type=All&status=All&langcode=All`
+  );
+
+  cy.get("tbody").then((tbody) => {
+    if (tbody.find("td.views-empty").length) {
+      cy.log("No branches to delete.");
+    } else {
+      cy.get('input[title="Select all rows in this table"]').check({
+        force: true,
+      });
+      cy.get("#edit-action").select("node_delete_action");
+      cy.contains("input", "Apply to selected items").click();
+      cy.contains("input", "Delete").click();
+    }
+  });
+};
+
+const visitOpeningHoursPage = () => {
+  const pageUrl = Cypress.env("pageUrl");
+  if (pageUrl) {
+    cy.visit(pageUrl);
+  }
+};
+
+const visitOpeningHoursPageAdmin = () => {
+  const adminUrl = Cypress.env("adminUrl");
+  if (adminUrl) {
+    cy.drupalLoginAndVisit(adminUrl);
+  }
+};
+
+const navigateToNextWeekAdmin = () => {
+  cy.get('button[title="Næste"]').click();
```

Comment:
It is a bit odd that we target something in Danish here when we usually work with English. I guess it is because we hardcode Fullcalendar to work in Danish.